ui/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -135,197 +98,271 @@ export default class ConfirmDecryptMessage extends Component {
           }`}
         </div>
       </div>
-    );
-  };
+    </div>
+  );
+};
 
-  renderRequestIcon = () => {
-    const { requesterAddress } = this.props;
+const VisualSection = ({ name, notice, targetSubjectMetadata }) => (
+  <div className="request-decrypt-message__visual">
+    <section>
+      {targetSubjectMetadata?.iconUrl ? (
+        <img
+          className="request-decrypt-message__visual-identicon"
+          src={targetSubjectMetadata.iconUrl}
+          alt=""
+        />
+      ) : (
+        <i className="request-decrypt-message__visual-identicon--default">
+          {name.charAt(0).toUpperCase()}
+        </i>
+      )}
+      <div className="request-decrypt-message__notice">{notice}</div>
+    </section>
+  </div>
+);
 
-    return (
-      <div className="request-decrypt-message__request-icon">
-        <Identicon diameter={40} address={requesterAddress} />
-      </div>
-    );
-  };
+const Footer = ({
+  cancelDecryptMessage,
+  clearConfirmTransaction,
+  decryptMessage,
+  hasScrolledToBottom,
+  history,
+  isScrollable,
+  mostRecentOverviewPage,
+  txData,
+}) => {
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
 
-  renderAccountInfo = () => {
-    return (
-      <div className="request-decrypt-message__account-info">
-        {this.renderAccount()}
-        {this.renderRequestIcon()}
-        {this.renderBalance()}
-      </div>
-    );
-  };
+  return (
+    <PageContainerFooter
+      cancelText={t('cancel')}
+      submitText={t('decrypt')}
+      disabled={isScrollable && !hasScrolledToBottom}
+      onCancel={async (event) => {
```

Comment:
Minor, is it worth a `onCancelClick` function for this given the size?

Reply:
Done